src/Compilers/CSharp/Portable/Symbols/OverriddenOrHiddenMembersHelpers.cs

```diff
@@ -131,11 +132,13 @@ private static OverriddenOrHiddenMembersResult MakeOverriddenOrHiddenMembersWork
             Symbol bestMatch = null;
             ArrayBuilder<Symbol> hiddenBuilder = null;
 
-            // If the metadata indicates a specific override, use that.
-            Symbol explicitOverride = member switch
+            // A specific override exact match candidate, if one is known. This supports covariant returns, for which signature
+            // matching is not sufficient. This member is treated as being as good as an exact match.
+            Symbol knownOverriddenMember = member switch
             {
-                PEMethodSymbol method => method.ExplicitlyOverriddenClassMethod,
-                PEPropertySymbol property => property.ExplicitlyOverriddenClassProperty,
+                MethodSymbol method => KnownOverriddenClassMethod(method),
+                PEPropertySymbol { GetMethod: PEMethodSymbol { ExplicitlyOverriddenClassMethod: { AssociatedSymbol: PropertySymbol overriddenProperty } } } => overriddenProperty,
+                RetargetingPropertySymbol property => property.RetargetOverriddenProperty,
```

Comment:
>property.RetargetOverriddenProperty [](start = 54, length = 35)

It feels like the logic here should be the same as for PEPropertySymbol above, I.e. infer from the getter

Reply:
If you have a valid retargeting scenario in mind for which this intermediate value results in incorrect compiler behavior, please identify that scenario. Otherwise I prefer it this way.

---
In reply to: [441992380](https://github.com/dotnet/roslyn/pull/44025#discussion_r441992380) [](ancestors = 441992380)